
<h1>ACTIVITY 1 - PALMER PENGUIN DATA ANALYTICS PROJECT </h1>
<hr>
<h3>ANALYST - LOUIEZZY RAMOS </h3>

In [1]:
import numpy as np
import pandas as pd
import warnings
warnings.filterwarnings('ignore')

<h3>COVERTING DATASET(S) INTO DATAFRAME </h3>
<h2>Requirements:</h2>
<ul>
    <li>
        Name
    </li>
</ul>


In [2]:
penguins = pd.read_csv('datasets\\penguins.csv')

In [3]:
penguins

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0       Adelie  Torgersen            39.1           18.7              181.0   
1       Adelie  Torgersen            39.5           17.4              186.0   
2       Adelie  Torgersen            40.3           18.0              195.0   
3       Adelie  Torgersen             NaN            NaN                NaN   
4       Adelie  Torgersen            36.7           19.3              193.0   
..         ...        ...             ...            ...                ...   
339  Chinstrap      Dream            55.8           19.8              207.0   
340  Chinstrap      Dream            43.5           18.1              202.0   
341  Chinstrap      Dream            49.6           18.2              193.0   
342  Chinstrap      Dream            50.8           19.0              210.0   
343  Chinstrap      Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
0         3750.0    male  2007  
1         3800.0  female  2007  
2         3250.0  female  2007  
3            NaN     NaN  2007  
4         3450.0  female  2007  
..           ...     ...   ...  
339       4000.0    male  2009  
340       3400.0  female  2009  
341       3775.0    male  2009  
342       4100.0    male  2009  
343       3775.0  female  2009  

[344 rows x 8 columns]

<h3>DESCRIPTIVE ANALYTICS</h3>

In [4]:
penguins.describe()

bill_length_mm  bill_depth_mm  flipper_length_mm  body_mass_g  \
count      342.000000     342.000000         342.000000   342.000000   
mean        43.921930      17.151170         200.915205  4201.754386   
std          5.459584       1.974793          14.061714   801.954536   
min         32.100000      13.100000         172.000000  2700.000000   
25%         39.225000      15.600000         190.000000  3550.000000   
50%         44.450000      17.300000         197.000000  4050.000000   
75%         48.500000      18.700000         213.000000  4750.000000   
max         59.600000      21.500000         231.000000  6300.000000   

              year  
count   344.000000  
mean   2008.029070  
std       0.818356  
min    2007.000000  
25%    2007.000000  
50%    2008.000000  
75%    2009.000000  
max    2009.000000

In [5]:
penguins.describe().T

count         mean         std     min       25%      50%  \
bill_length_mm     342.0    43.921930    5.459584    32.1    39.225    44.45   
bill_depth_mm      342.0    17.151170    1.974793    13.1    15.600    17.30   
flipper_length_mm  342.0   200.915205   14.061714   172.0   190.000   197.00   
body_mass_g        342.0  4201.754386  801.954536  2700.0  3550.000  4050.00   
year               344.0  2008.029070    0.818356  2007.0  2007.000  2008.00   

                      75%     max  
bill_length_mm       48.5    59.6  
bill_depth_mm        18.7    21.5  
flipper_length_mm   213.0   231.0  
body_mass_g        4750.0  6300.0  
year               2009.0  2009.0

<h3>CHECKING FOR NULL DATA</h3>

In [6]:
penguins.isnull()

species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0      False   False           False          False              False   
1      False   False           False          False              False   
2      False   False           False          False              False   
3      False   False            True           True               True   
4      False   False           False          False              False   
..       ...     ...             ...            ...                ...   
339    False   False           False          False              False   
340    False   False           False          False              False   
341    False   False           False          False              False   
342    False   False           False          False              False   
343    False   False           False          False              False   

     body_mass_g    sex   year  
0          False  False  False  
1          False  False  False  
2          False  False  False  
3           True   True  False  
4          False  False  False  
..           ...    ...    ...  
339        False  False  False  
340        False  False  False  
341        False  False  False  
342        False  False  False  
343        False  False  False  

[344 rows x 8 columns]

In [7]:
penguins.isnull().sum()

species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

In [8]:
penguins['species']

0         Adelie
1         Adelie
2         Adelie
3         Adelie
4         Adelie
         ...    
339    Chinstrap
340    Chinstrap
341    Chinstrap
342    Chinstrap
343    Chinstrap
Name: species, Length: 344, dtype: object

In [9]:
penguins['species'].value_counts()

species
Adelie       152
Gentoo       124
Chinstrap     68
Name: count, dtype: int64

In [10]:
penguins[penguins['species'] == 'Chinstrap']

species island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
276  Chinstrap  Dream            46.5           17.9              192.0   
277  Chinstrap  Dream            50.0           19.5              196.0   
278  Chinstrap  Dream            51.3           19.2              193.0   
279  Chinstrap  Dream            45.4           18.7              188.0   
280  Chinstrap  Dream            52.7           19.8              197.0   
..         ...    ...             ...            ...                ...   
339  Chinstrap  Dream            55.8           19.8              207.0   
340  Chinstrap  Dream            43.5           18.1              202.0   
341  Chinstrap  Dream            49.6           18.2              193.0   
342  Chinstrap  Dream            50.8           19.0              210.0   
343  Chinstrap  Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
276       3500.0  female  2007  
277       3900.0    male  2007  
278       3650.0    male  2007  
279       3525.0  female  2007  
280       3725.0    male  2007  
..           ...     ...   ...  
339       4000.0    male  2009  
340       3400.0  female  2009  
341       3775.0    male  2009  
342       4100.0    male  2009  
343       3775.0  female  2009  

[68 rows x 8 columns]

In [11]:
penguins[penguins['species'] == 'Gentoo']

species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
152  Gentoo  Biscoe            46.1           13.2              211.0   
153  Gentoo  Biscoe            50.0           16.3              230.0   
154  Gentoo  Biscoe            48.7           14.1              210.0   
155  Gentoo  Biscoe            50.0           15.2              218.0   
156  Gentoo  Biscoe            47.6           14.5              215.0   
..      ...     ...             ...            ...                ...   
271  Gentoo  Biscoe             NaN            NaN                NaN   
272  Gentoo  Biscoe            46.8           14.3              215.0   
273  Gentoo  Biscoe            50.4           15.7              222.0   
274  Gentoo  Biscoe            45.2           14.8              212.0   
275  Gentoo  Biscoe            49.9           16.1              213.0   

     body_mass_g     sex  year  
152       4500.0  female  2007  
153       5700.0    male  2007  
154       4450.0  female  2007  
155       5700.0    male  2007  
156       5400.0    male  2007  
..           ...     ...   ...  
271          NaN     NaN  2009  
272       4850.0  female  2009  
273       5750.0    male  2009  
274       5200.0  female  2009  
275       5400.0    male  2009  

[124 rows x 8 columns]

In [12]:
penguins[penguins['species'] == 'Adelie']

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0    Adelie  Torgersen            39.1           18.7              181.0   
1    Adelie  Torgersen            39.5           17.4              186.0   
2    Adelie  Torgersen            40.3           18.0              195.0   
3    Adelie  Torgersen             NaN            NaN                NaN   
4    Adelie  Torgersen            36.7           19.3              193.0   
..      ...        ...             ...            ...                ...   
147  Adelie      Dream            36.6           18.4              184.0   
148  Adelie      Dream            36.0           17.8              195.0   
149  Adelie      Dream            37.8           18.1              193.0   
150  Adelie      Dream            36.0           17.1              187.0   
151  Adelie      Dream            41.5           18.5              201.0   

     body_mass_g     sex  year  
0         3750.0    male  2007  
1         3800.0  female  2007  
2         3250.0  female  2007  
3            NaN     NaN  2007  
4         3450.0  female  2007  
..           ...     ...   ...  
147       3475.0  female  2009  
148       3450.0  female  2009  
149       3750.0    male  2009  
150       3700.0  female  2009  
151       4000.0    male  2009  

[152 rows x 8 columns]

In [13]:
Gentoo_df = penguins[penguins['species'] == 'Gentoo']

In [14]:
Gentoo_df

species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
152  Gentoo  Biscoe            46.1           13.2              211.0   
153  Gentoo  Biscoe            50.0           16.3              230.0   
154  Gentoo  Biscoe            48.7           14.1              210.0   
155  Gentoo  Biscoe            50.0           15.2              218.0   
156  Gentoo  Biscoe            47.6           14.5              215.0   
..      ...     ...             ...            ...                ...   
271  Gentoo  Biscoe             NaN            NaN                NaN   
272  Gentoo  Biscoe            46.8           14.3              215.0   
273  Gentoo  Biscoe            50.4           15.7              222.0   
274  Gentoo  Biscoe            45.2           14.8              212.0   
275  Gentoo  Biscoe            49.9           16.1              213.0   

     body_mass_g     sex  year  
152       4500.0  female  2007  
153       5700.0    male  2007  
154       4450.0  female  2007  
155       5700.0    male  2007  
156       5400.0    male  2007  
..           ...     ...   ...  
271          NaN     NaN  2009  
272       4850.0  female  2009  
273       5750.0    male  2009  
274       5200.0  female  2009  
275       5400.0    male  2009  

[124 rows x 8 columns]

In [15]:
Adelie_df = penguins[penguins['species'] == 'Adelie']

In [16]:
Adelie_df

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0    Adelie  Torgersen            39.1           18.7              181.0   
1    Adelie  Torgersen            39.5           17.4              186.0   
2    Adelie  Torgersen            40.3           18.0              195.0   
3    Adelie  Torgersen             NaN            NaN                NaN   
4    Adelie  Torgersen            36.7           19.3              193.0   
..      ...        ...             ...            ...                ...   
147  Adelie      Dream            36.6           18.4              184.0   
148  Adelie      Dream            36.0           17.8              195.0   
149  Adelie      Dream            37.8           18.1              193.0   
150  Adelie      Dream            36.0           17.1              187.0   
151  Adelie      Dream            41.5           18.5              201.0   

     body_mass_g     sex  year  
0         3750.0    male  2007  
1         3800.0  female  2007  
2         3250.0  female  2007  
3            NaN     NaN  2007  
4         3450.0  female  2007  
..           ...     ...   ...  
147       3475.0  female  2009  
148       3450.0  female  2009  
149       3750.0    male  2009  
150       3700.0  female  2009  
151       4000.0    male  2009  

[152 rows x 8 columns]

In [17]:
Chinstrap_df = penguins[penguins['species'] == 'Chinstrap']

In [18]:
Chinstrap_df

species island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
276  Chinstrap  Dream            46.5           17.9              192.0   
277  Chinstrap  Dream            50.0           19.5              196.0   
278  Chinstrap  Dream            51.3           19.2              193.0   
279  Chinstrap  Dream            45.4           18.7              188.0   
280  Chinstrap  Dream            52.7           19.8              197.0   
..         ...    ...             ...            ...                ...   
339  Chinstrap  Dream            55.8           19.8              207.0   
340  Chinstrap  Dream            43.5           18.1              202.0   
341  Chinstrap  Dream            49.6           18.2              193.0   
342  Chinstrap  Dream            50.8           19.0              210.0   
343  Chinstrap  Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
276       3500.0  female  2007  
277       3900.0    male  2007  
278       3650.0    male  2007  
279       3525.0  female  2007  
280       3725.0    male  2007  
..           ...     ...   ...  
339       4000.0    male  2009  
340       3400.0  female  2009  
341       3775.0    male  2009  
342       4100.0    male  2009  
343       3775.0  female  2009  

[68 rows x 8 columns]

In [19]:
penguins[penguins['species'] != 'Chinstrap']

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0    Adelie  Torgersen            39.1           18.7              181.0   
1    Adelie  Torgersen            39.5           17.4              186.0   
2    Adelie  Torgersen            40.3           18.0              195.0   
3    Adelie  Torgersen             NaN            NaN                NaN   
4    Adelie  Torgersen            36.7           19.3              193.0   
..      ...        ...             ...            ...                ...   
271  Gentoo     Biscoe             NaN            NaN                NaN   
272  Gentoo     Biscoe            46.8           14.3              215.0   
273  Gentoo     Biscoe            50.4           15.7              222.0   
274  Gentoo     Biscoe            45.2           14.8              212.0   
275  Gentoo     Biscoe            49.9           16.1              213.0   

     body_mass_g     sex  year  
0         3750.0    male  2007  
1         3800.0  female  2007  
2         3250.0  female  2007  
3            NaN     NaN  2007  
4         3450.0  female  2007  
..           ...     ...   ...  
271          NaN     NaN  2009  
272       4850.0  female  2009  
273       5750.0    male  2009  
274       5200.0  female  2009  
275       5400.0    male  2009  

[276 rows x 8 columns]

<h2>.INFO ni Anes</h2>

In [20]:
Gentoo_df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 124 entries, 152 to 275
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   species            124 non-null    object 
 1   island             124 non-null    object 
 2   bill_length_mm     123 non-null    float64
 3   bill_depth_mm      123 non-null    float64
 4   flipper_length_mm  123 non-null    float64
 5   body_mass_g        123 non-null    float64
 6   sex                119 non-null    object 
 7   year               124 non-null    int64  
dtypes: float64(4), int64(1), object(3)
memory usage: 8.7+ KB


<h2>DATARFRAME NI ATTECCO</h2>

In [21]:
(penguins['species'] == 'Chinstrap') & (penguins['body_mass_g'] >= 4000)

0      False
1      False
2      False
3      False
4      False
       ...  
339     True
340    False
341    False
342     True
343    False
Length: 344, dtype: bool

In [22]:
penguins[penguins['body_mass_g'] >= 4000]

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
7       Adelie  Torgersen            39.2           19.6              195.0   
9       Adelie  Torgersen            42.0           20.2              190.0   
14      Adelie  Torgersen            34.6           21.1              198.0   
17      Adelie  Torgersen            42.5           20.7              197.0   
19      Adelie  Torgersen            46.0           21.5              194.0   
..         ...        ...             ...            ...                ...   
323  Chinstrap      Dream            49.0           19.6              212.0   
329  Chinstrap      Dream            50.7           19.7              203.0   
333  Chinstrap      Dream            49.3           19.9              203.0   
339  Chinstrap      Dream            55.8           19.8              207.0   
342  Chinstrap      Dream            50.8           19.0              210.0   

     body_mass_g   sex  year  
7         4675.0  male  2007  
9         4250.0   NaN  2007  
14        4400.0  male  2007  
17        4500.0  male  2007  
19        4200.0  male  2007  
..           ...   ...   ...  
323       4300.0  male  2009  
329       4050.0  male  2009  
333       4050.0  male  2009  
339       4000.0  male  2009  
342       4100.0  male  2009  

[177 rows x 8 columns]

<h2>FILLING UP NULL VALUES</h2>

In [23]:
penguins.isnull().sum()

species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

In [24]:
penguins['bill_depth_mm'].mean()

np.float64(17.151169590643274)

In [25]:
penguins['bill_depth_mm'] = penguins['bill_depth_mm'].fillna(penguins['bill_depth_mm'].mean())

In [26]:
penguins.isnull().sum()

species               0
island                0
bill_length_mm        2
bill_depth_mm         0
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

In [27]:
penguins['flipper_length_mm'].mean()

np.float64(200.91520467836258)

In [28]:
penguins['flipper_length_mm'] = penguins['flipper_length_mm'].fillna(penguins['flipper_length_mm'].mean())

In [29]:
penguins.isnull().sum()

species               0
island                0
bill_length_mm        2
bill_depth_mm         0
flipper_length_mm     0
body_mass_g           2
sex                  11
year                  0
dtype: int64

In [30]:
penguins['body_mass_g'].mean()

np.float64(4201.754385964912)

In [31]:
penguins['body_mass_g'] = penguins['body_mass_g'].fillna(penguins['body_mass_g'].mean())

In [32]:
penguins.isnull().sum()

species               0
island                0
bill_length_mm        2
bill_depth_mm         0
flipper_length_mm     0
body_mass_g           0
sex                  11
year                  0
dtype: int64

In [33]:
penguins['sex'].mode()

0    male
Name: sex, dtype: object

In [34]:
penguins['sex'] = penguins['sex'].fillna(penguins['sex'].mode()[0])

In [35]:
penguins

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0       Adelie  Torgersen            39.1       18.70000         181.000000   
1       Adelie  Torgersen            39.5       17.40000         186.000000   
2       Adelie  Torgersen            40.3       18.00000         195.000000   
3       Adelie  Torgersen             NaN       17.15117         200.915205   
4       Adelie  Torgersen            36.7       19.30000         193.000000   
..         ...        ...             ...            ...                ...   
339  Chinstrap      Dream            55.8       19.80000         207.000000   
340  Chinstrap      Dream            43.5       18.10000         202.000000   
341  Chinstrap      Dream            49.6       18.20000         193.000000   
342  Chinstrap      Dream            50.8       19.00000         210.000000   
343  Chinstrap      Dream            50.2       18.70000         198.000000   

     body_mass_g     sex  year  
0    3750.000000    male  2007  
1    3800.000000  female  2007  
2    3250.000000  female  2007  
3    4201.754386    male  2007  
4    3450.000000  female  2007  
..           ...     ...   ...  
339  4000.000000    male  2009  
340  3400.000000  female  2009  
341  3775.000000    male  2009  
342  4100.000000    male  2009  
343  3775.000000  female  2009  

[344 rows x 8 columns]

penguins.isnull().sum()

In [36]:
penguins.columns

Index(['species', 'island', 'bill_length_mm', 'bill_depth_mm',
       'flipper_length_mm', 'body_mass_g', 'sex', 'year'],
      dtype='object')

In [37]:
penguins['mass_kg'] = penguins['body_mass_g']/1000

In [38]:
penguins.columns

Index(['species', 'island', 'bill_length_mm', 'bill_depth_mm',
       'flipper_length_mm', 'body_mass_g', 'sex', 'year', 'mass_kg'],
      dtype='object')

In [39]:
penguins['mass_kg'].max()

np.float64(6.3)

<h2 style= 'text-align: center'> DATA ANALYTICS - ACTIVITY 1 - RAMOS </h2>

<hr> 
<h2 style="color:gray"> Q1: How many male Gentoo penguins are there? </h2>

In [40]:
gentoo_male = penguins[penguins['species'] == 'Gentoo']
gentoo_male

species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
152  Gentoo  Biscoe            46.1       13.20000         211.000000   
153  Gentoo  Biscoe            50.0       16.30000         230.000000   
154  Gentoo  Biscoe            48.7       14.10000         210.000000   
155  Gentoo  Biscoe            50.0       15.20000         218.000000   
156  Gentoo  Biscoe            47.6       14.50000         215.000000   
..      ...     ...             ...            ...                ...   
271  Gentoo  Biscoe             NaN       17.15117         200.915205   
272  Gentoo  Biscoe            46.8       14.30000         215.000000   
273  Gentoo  Biscoe            50.4       15.70000         222.000000   
274  Gentoo  Biscoe            45.2       14.80000         212.000000   
275  Gentoo  Biscoe            49.9       16.10000         213.000000   

     body_mass_g     sex  year   mass_kg  
152  4500.000000  female  2007  4.500000  
153  5700.000000    male  2007  5.700000  
154  4450.000000  female  2007  4.450000  
155  5700.000000    male  2007  5.700000  
156  5400.000000    male  2007  5.400000  
..           ...     ...   ...       ...  
271  4201.754386    male  2009  4.201754  
272  4850.000000  female  2009  4.850000  
273  5750.000000    male  2009  5.750000  
274  5200.000000  female  2009  5.200000  
275  5400.000000    male  2009  5.400000  

[124 rows x 9 columns]

In [41]:
gentoo_male['sex'].value_counts()

sex
male      66
female    58
Name: count, dtype: int64

<h3 style="color:blue">INSIGHTS: It shows the domination of male penguins over female penguins, with a count of 66 compared to 58.</h3>
<hr> 


<h2 style="color:gray"> Q2: How many penguins per island? </h2>


In [42]:
p_island_counts = penguins['island'].value_counts()
p_island_counts

island
Biscoe       168
Dream        124
Torgersen     52
Name: count, dtype: int64

<h3 style="color:blue">INSIGHTS: The record shows, Biscoe Island is home to the highest number of penguins, with a total of 168, followed by Dream Island with 124. Torgersen Island has the fewest penguins, with only 52 recorded.</h3>
<hr> 

<h2 style="color:gray"> Q3: Which sex is heavier in Chinstrap species?</h2>

In [43]:
chinstrap_body_mass = np.round(penguins.groupby(["species", "sex"])["mass_kg"].mean(), 1)
chinstrap_body_mass

species    sex   
Adelie     female    3.4
           male      4.0
Chinstrap  female    3.5
           male      3.9
Gentoo     female    4.7
           male      5.4
Name: mass_kg, dtype: float64

<h3 style="color:blue"> INSIGHTS: The table shows that male Chinstrap species, weighing 3.9kg, are heavier than females.</h3>
<hr> 

<h2 style="color:gray">Q4: What is the average weight of Gentoo species? </h2>

In [44]:
gentoo_ave_bmass = np.round(penguins['mass_kg'].mean(), 1)
gentoo_ave_bmass

np.float64(4.2)

<h3 style="color:blue">INSIGHTS: The average weight of the Gentoo species is 4.2kg. </h3>
<hr> 

<h2 style="color:gray"> Q5: What is the median flipper length? </h2>

In [45]:
median_flipper = penguins['flipper_length_mm'].median()
median_flipper

np.float64(197.0)

<h3 style="color:blue"> INSIGHTS: It shows that 197mm is the median flipper length. </h3>
<hr> 

<h2 style="color:gray"> Q6: What is the most common species on Biscoe Island? </h2>

In [46]:
biscoe_penguins = penguins[penguins['island'] == 'Biscoe']
biscoe_penguins

species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
20   Adelie  Biscoe            37.8       18.30000         174.000000   
21   Adelie  Biscoe            37.7       18.70000         180.000000   
22   Adelie  Biscoe            35.9       19.20000         189.000000   
23   Adelie  Biscoe            38.2       18.10000         185.000000   
24   Adelie  Biscoe            38.8       17.20000         180.000000   
..      ...     ...             ...            ...                ...   
271  Gentoo  Biscoe             NaN       17.15117         200.915205   
272  Gentoo  Biscoe            46.8       14.30000         215.000000   
273  Gentoo  Biscoe            50.4       15.70000         222.000000   
274  Gentoo  Biscoe            45.2       14.80000         212.000000   
275  Gentoo  Biscoe            49.9       16.10000         213.000000   

     body_mass_g     sex  year   mass_kg  
20   3400.000000  female  2007  3.400000  
21   3600.000000    male  2007  3.600000  
22   3800.000000  female  2007  3.800000  
23   3950.000000    male  2007  3.950000  
24   3800.000000    male  2007  3.800000  
..           ...     ...   ...       ...  
271  4201.754386    male  2009  4.201754  
272  4850.000000  female  2009  4.850000  
273  5750.000000    male  2009  5.750000  
274  5200.000000  female  2009  5.200000  
275  5400.000000    male  2009  5.400000  

[168 rows x 9 columns]

In [47]:
biscoe_species_mode = (biscoe_penguins['species'].mode()[0])
biscoe_species_mode

'Gentoo'

<h3 style="color:blue">INSIGHTS: In Biscoe Island, Gentoo penguins are common. </h3>
<hr> 

<h2 style="color:gray">Q7: How many female Adelie penguins are recorded? </h2>

In [48]:
adelie_female = penguins[penguins['species'] == 'Adelie']
adelie_female

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0    Adelie  Torgersen            39.1       18.70000         181.000000   
1    Adelie  Torgersen            39.5       17.40000         186.000000   
2    Adelie  Torgersen            40.3       18.00000         195.000000   
3    Adelie  Torgersen             NaN       17.15117         200.915205   
4    Adelie  Torgersen            36.7       19.30000         193.000000   
..      ...        ...             ...            ...                ...   
147  Adelie      Dream            36.6       18.40000         184.000000   
148  Adelie      Dream            36.0       17.80000         195.000000   
149  Adelie      Dream            37.8       18.10000         193.000000   
150  Adelie      Dream            36.0       17.10000         187.000000   
151  Adelie      Dream            41.5       18.50000         201.000000   

     body_mass_g     sex  year   mass_kg  
0    3750.000000    male  2007  3.750000  
1    3800.000000  female  2007  3.800000  
2    3250.000000  female  2007  3.250000  
3    4201.754386    male  2007  4.201754  
4    3450.000000  female  2007  3.450000  
..           ...     ...   ...       ...  
147  3475.000000  female  2009  3.475000  
148  3450.000000  female  2009  3.450000  
149  3750.000000    male  2009  3.750000  
150  3700.000000  female  2009  3.700000  
151  4000.000000    male  2009  4.000000  

[152 rows x 9 columns]

In [49]:
adelie_female['sex'].value_counts()

sex
male      79
female    73
Name: count, dtype: int64

<h3 style="color:blue">INSIGHTS: The record shows that the Adelie species has 73 penguins. </h3>
<hr> 

<h2 style="color:gray">Q8: Do male penguins weigh more than female penguins? </h2>

In [50]:
mass_by_sex = penguins.groupby('sex')['mass_kg'].mean()
mass_by_sex

sex
female    3.862273
male      4.514684
Name: mass_kg, dtype: float64

<h3 style="color:blue">INSIGHTS: Male penguins' average body mass weighs 4.5kg, compared to female penguins' body mass average 3.9kg. </h3>
<hr> 

<h2 style="color:gray">Q9: Which species has the largest difference in body mass between males and females? </h2>

In [51]:
mass_species_sex = penguins.groupby(['species','sex'])['mass_kg'].mean()
mass_species_sex

species    sex   
Adelie     female    3.368836
           male      4.013630
Chinstrap  female    3.527206
           male      3.938971
Gentoo     female    4.679741
           male      5.411011
Name: mass_kg, dtype: float64

<h3 style="color:blue">INSIGHTS: Gentoo penguins have the biggest difference in body mass between males and females, with males being heavier. Adelie and Chinstrap penguins show smaller differences, meaning their sizes are more balanced.</h3>
<hr> 

<h2 style="color:gray">Q10: Which year had the smallest average bill depth?</h2>

In [53]:
bill_year = penguins.groupby('year')['bill_depth_mm'].mean()
bill_year

year
2007    17.425011
2008    16.914035
2009    17.125426
Name: bill_depth_mm, dtype: float64

<h3 style="color:blue">INSIGHTS: The year 2008 had the smallest average bill depth among the recorded penguins. </h3>
<hr> 

<h2 style="color:gray">Q11: Do penguins with longer flippers also tend to have longer bills? </h2>

In [54]:
corr_flipper_bill = penguins[['flipper_length_mm','bill_length_mm']].corr()
corr_flipper_bill

flipper_length_mm  bill_length_mm
flipper_length_mm           1.000000        0.656181
bill_length_mm              0.656181        1.000000

<h3 style="color:blue">INSIGHTS: This means penguins with longer flippers also tend to have longer bills, although it’s not a perfect relationship. </h3>
<hr> 

<h2 style="color:gray">Q12: Which species has the most balanced male-to-female ratio?</h2>

In [55]:
sex_balance = penguins.groupby('species')['sex'].value_counts()
sex_balance

species    sex   
Adelie     male      79
           female    73
Chinstrap  female    34
           male      34
Gentoo     male      66
           female    58
Name: count, dtype: int64

<h3 style="color:blue">INSIGHTS: The Chinstrap penguins have the most balanced male-to-female ratio, with an equal number of males and females. </h3>
<hr> 

<h2 style="color:gray">Q13: Which year recorded the heaviest penguins overall? </h2>

In [57]:
mass_year = penguins.groupby('year')['mass_kg'].mean()
mass_year

year
2007    4.125243
2008    4.266667
2009    4.210223
Name: mass_kg, dtype: float64

<h3 style="color:blue">INSIGHTS: The differences between each year are close to each other. However, it has been recorded that in 2008, the average body mass was 4.27 kg, while in 2009 it was 4.21 kg.  </h3>
<hr> 